pkgs/development/python-modules/tkinter/default.nix

```diff
@@ -1,34 +1,62 @@
 {
   lib,
-  stdenv,
   buildPythonPackage,
+  setuptools,
   python,
-  py,
-  isPyPy,
+  tcl-9_0,
+  tk-9_0,
 }:
 
+let
+  tcl = tcl-9_0;
+  tk = tk-9_0;
+in
+
 buildPythonPackage {
   pname = "tkinter";
   version = python.version;
-  src = py;
-  format = "other";
-
-  # tkinter is included in PyPy, making this package a no-op.
-  installPhase = lib.optionalString (!isPyPy) (
-    ''
-      # Move the tkinter module
-      mkdir -p $out/${py.sitePackages}
-      mv lib/${py.libPrefix}/lib-dynload/_tkinter* $out/${py.sitePackages}/
-    ''
-    + lib.optionalString (!stdenv.hostPlatform.isDarwin) ''
-      # Update the rpath to point to python without x11Support
-      old_rpath=$(patchelf --print-rpath $out/${py.sitePackages}/_tkinter*)
-      new_rpath=$(sed "s#${py}#${python}#g" <<< "$old_rpath" )
-      patchelf --set-rpath $new_rpath $out/${py.sitePackages}/_tkinter*
-    ''
-  );
-
-  meta = py.meta // {
+  pyproject = true;
+
+  src = python.src;
+
+  postPatch = ''
+    mkdir $NIX_BUILD_TOP/tkinter
+
+    cp -v  Modules/{_tkinter.c,tkinter.h} ../tkinter/
+    cp -rv Modules/clinic ../tkinter/
```

Comment:
Maybe we should expose the Argument Clinic as an output of the Python derivation or something? They do say it’s internal‐only, but also that “if you maintain an external C extension for CPython, you’re welcome to experiment with Argument Clinic in your own code”.

Reply:
I don't fancy exposing things for the thrill of it. It wasn't needed until now, so I won't  bother.